### DISCUSSION TITLE
Mixing MartenDB and EF

### DISCUSSION THREAD

**Discussion by nkosi23** [JasperFx/marten#2611] [Q&A]
Hello,

As someone totally new to Marten (and Postgresql for the matter), one thing that is not clear to me by reading the docs is: how can we interact directly with Postgresql (for example using tools such as EF or Npgsql), without risking introducing state management conflicts. To put it differently, I am not clear on the extent to which MartenDB "takes control" of our database and expects us to manage things through it to function properly.

My idea/goal would be to use MartenDB for Document and Event Sourcing needs, and an O or for relational data, on the same database. I wasn't able to find a section of the documentation discussing such a usage, even though the docs allude to the fact that developers keep having access to all postgres features.

Thanks

**Comment by gfoidl** [comment]
Marten can be [configured](https://martendb.io/schema/migrations.html) on what to do with the DB-schema.
Basically it cares only about the schema-elements it creates by itself, and leaves other schema-elements alone. So if you, EF, anything else creates schema-elements in the DB, so marten won't touch them (by default).

From this point of view I don't see a problem using Marten + EF together.
But I'm unsure by the point how EF migrations will treat the Marten created schema-elements. The best is to just build a demo app and try it out.
